sdk/turing/router/router.py

```diff
@@ -47,8 +48,10 @@ def __init__(self,
 
         if config is not None:
             self._config = RouterConfig(name=name, environment_name=environment_name, **config)
+            self._version = config.get('version')
```

Comment:
I think I probably missed it in the [original PR](https://github.com/gojek/turing/pull/171).

Why can't the implementation just be:

```python
    @property
    def version(self) -> int:
        return self._config.get('version') if self._config or None
```

Why do we need the `version` argument on init?

Reply:
You're right, there's no need for version to be up there in the constructor.

On the other hand, we can't just remove the second line in this `if` conditional and replace it with the getter as you've suggested, because `self._config`, which is a `RouterConfig` instance, does not contain a `version` attribute:
```python
 if config is not None:
            self._config = RouterConfig(name=name, environment_name=environment_name, **config)
            self._version = config.get('version')
```
Instead, we need to extract the `version` from the `config` argument, which is only available when the `Router` instance is being constructed.

Reply:
Actually, now that I think of it, I might just refactor the `config` getter this way:
```python
 @property
    def config(self) -> `RouterConfig`:
        return RouterConfig(name=self.name, environment_name=self.environment_name, **self._config)
```
Doing so will allow us to keep the `version` getter that you've suggested, as well as remove the awkward if conditional in the constructor.

Reply:
In short, we'll get something like:

```python
...
   def __init__(self,
                 id: int,
                 name: str,
                 project_id: int,
                 environment_name: str,
                 monitoring_url: str,
                 status: str,
                 version: int = None,
                 config: Dict = None,
                 endpoint: str = None,
                 **kwargs):
        super(Router, self).__init__(**kwargs)
        self._id = id
        self._name = name
        self._project_id = project_id
        self._environment_name = environment_name
        self._endpoint = endpoint
        self._monitoring_url = monitoring_url
        self._status = RouterStatus(status)
        self._config = config
...
 @property
    def config(self) -> `RouterConfig`:
        return RouterConfig(name=self.name, environment_name=self.environment_name, **self._config)

@property
def version(self) -> int:
    return self._config.get('version') if self._config else None
```

Reply:
Ah right. Makes sense to me.